File: app/src/main/java/com/konifar/materialcat/views/ShareBarView.java
package com.konifar.materialcat.views;

import android.app.Activity;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;

import com.konifar.materialcat.R;
import com.konifar.materialcat.utils.ShareUtils;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class ShareBarView extends LinearLayout {

    public ShareBarView(Context context) {
        super(context);
    }

    public ShareBarView(Context context, AttributeSet attrs) {
        super(context, attrs);
        inflate(context, R.layout.ui_share_bar, this);
        ButterKnife.bind(this);
    }


    @OnClick(R.id.btn_email)
    void onClickBtnEmail() {
        //
    }

    @OnClick(R.id.btn_copy)
    void onClickBtnCopy() {
        //
    }

    @OnClick(R.id.btn_google_plus)
    void onClickBtnGooglePlus() {
        //
    }

    @OnClick(R.id.btn_facebook)
    void onClickBtnFacebook() {
        ShareUtils.showShareDialog((Activity) getContext());
    }

    @OnClick(R.id.btn_twitter)
    void onClickTwitter() {
        //
    }

}

File: app/src/main/java/com/konifar/materialcat/views/ShareBarView.kt
package com.konifar.materialcat.views

import android.app.Activity
import android.content.ClipData
import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import butterknife.ButterKnife
import butterknife.OnClick
import com.konifar.materialcat.R
import com.konifar.materialcat.utils.AppUtils
import com.konifar.materialcat.utils.ShareUtils

class ShareBarView : LinearLayout {

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        View.inflate(context, R.layout.ui_share_bar, this)
        ButterKnife.bind(this)
    }


    @OnClick(R.id.btn_email)
    internal fun onClickBtnEmail() {
        val mails = arrayOfNulls<String>(1)
        AppUtils.openMailChooser(context, ShareUtils.REPOGITORY_URL, mails,
                context.getString(R.string.app_name))
    }

    @OnClick(R.id.btn_copy)
    internal fun onClickBtnCopy() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
            val clip = ClipData.newPlainText("simple text", ShareUtils.REPOGITORY_URL)
            AppUtils.showToast(R.string.copyed, context)
        } else {
            AppUtils.showToast(R.string.not_supported, context)
        }
    }

    @OnClick(R.id.btn_google_plus)
    internal fun onClickBtnGooglePlus() {
        // TODO
        AppUtils.showToast(R.string.coming_soon, context)
    }

    @OnClick(R.id.btn_facebook)
    internal fun onClickBtnFacebook() {
        ShareUtils.showShareDialog(context as Activity)
    }

    @OnClick(R.id.btn_twitter)
    internal fun onClickTwitter() {
        // TODO
        AppUtils.showToast(R.string.coming_soon, context)
    }

}
